libvirt/container-linux/kubernetes/libvirt.tf

```diff
@@ -0,0 +1,38 @@
+// Some basic libvirt building blocks:
+// - the network
+// - the base QCOW volume 
+
+locals {
+  network_name = "${var.libvirt_network_name == "" ? var.cluster_name : var.libvirt_network_name}"
+}
+
+resource "libvirt_network" "net" {
+  name = "${local.network_name}"
+
+  mode   = "nat"
+  bridge = "${var.libvirt_network_if}"
+
+  domain = "${var.domain}"
+
+  addresses = ["${var.node_ip_pool}"]
+
+  dns_forwarder {
+    address = "${var.dns_server}"
+  }
+}
+
+resource "libvirt_volume" "coreos_base" {
+  name   = "coreos_base"
+  pool   = "${var.libvirt_storage_pool}"
+  source = "${var.base_image_path}"
+}
+
+# Set up the cluster domain name
+# This is currently limited to the first controller, due to an issue with net-update, even though libvirt supports multiple a-records
+resource "null_resource" "k8s_dns" {
+  provisioner "local-exec" {
+    command = "virsh -c ${var.libvirt_endpoint} net-update ${local.network_name} add dns-host \"<host ip='${var.controller_ips[0]}'><hostname>${var.k8s_hostname}</hostname></host>\" --live --config"
```

Comment:
Don't like this :/

Reply:
I absolutely hate this. But `terraform-provider-libvirt` doesn't support extra domain names, and the PR to merge it (dmacvicar/terraform-provider-libvirt#288) is stalled.

I would love to have the user do this manually, but it can only be done after the libvirt network is created and before the bootkube process starts.